python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -53,76 +66,108 @@ def __init__(self):
         ret_val_doc += "For the 'update_blocked' status, data will contain a reason key containing an explanation."
 
         self.parameters = {"return_value": {"description": ret_val_doc, "type": "dict" }}
-            
+
+    def _parse_arguments(self, parameters):
+        """
+        Parses the list of arguments from the command line.
+
+        :param parameters: The content of argv that hasn't been processed by the tank command.
+
+        :returns: The core version. None if --version wasn't specified.
+        """
+        parser = TkOptParse()
+        parser.set_usage(optparse.SUPPRESS_USAGE)
+        parser.add_option("-v", "--version", type="string", default=None)
+        options, args = parser.parse_args()
```

Comment:
you previously mentioned backwards compatibility - previously someone running `./tank core --foo --bar` would be fine, after the upgrade they will get an error. Something we should be worried about?